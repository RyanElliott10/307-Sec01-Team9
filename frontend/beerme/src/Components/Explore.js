import React, { Component } from "react";
import ExploreButton from "./ExploreButton";
import Header from "./Layout/Header";

export class Explore extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  componentDidMount() {
    fetch("http://jsonplaceholder.typicode.com/todos")
      .then(res => res.json())
      .then(data => {
        data.forEach(element => {
          element.selected = false;
        });
        this.setState({ contacts: data.slice(0, 20) });
      })
      .catch(console.log);
  }

  onPreviousClick = () => {
    console.log("onPreviousClick");
  };

  onNextClick = () => {
    const selections = this.state.contacts.filter(data => data.selected);
    console.log(selections);
  };

  _onCheckboxClick = id => {
    this.setState({
      data: this.state.contacts.map(option => {
        if (option.id === id) {
          option.selected = !option.selected;
        }
        return option;
      })
    });
  };

  _renderHeader() {
    return <Header />;
  }

  _renderTopBlurb() {
    return (
      <React.Fragment>
        <h1 style={titleStyle}>Explore</h1>
        <h5 style={descStyle}>
          Use our unique beer exploration tool to find new and interesting beers
          for you!
        </h5>
      </React.Fragment>
    );
  }

  _renderSelectionBox() {
    return (
      <div style={selectionBoxStyle}>
        {this._renderSelBoxTopText()}
        {this._renderSelections()}
      </div>
    );
  }

  _renderSelBoxTopText() {
    return (
      <div style={{ flex: 1, justifyContent: "center", alignItems: "center" }}>
        <h2>Preferred Notes</h2>
        <h5>This is where the explanation of what a note is in beer lives.</h5>
      </div>
    );
  }

  _renderSelections() {
    if (this.state.contacts) {
      return (
        <React.Fragment>
          {this.state.contacts.map(data => this._renderChecboxOption(data))}
        </React.Fragment>
      );
    }
    return null;
  }

  _renderChecboxOption(data) {
    return (
      <div key={data.id}>
        <input
          type="checkbox"
          onClick={this._onCheckboxClick.bind(this, data.id)}
        />
        {data.title}
      </div>
    );
  }

  _renderProgressionButtons() {
    return (
      <div style={btnsStyle}>
        <ExploreButton
          style={{ paddingTop: "10px" }}
          onClick={this.onPreviousClick}
          title={"<-- Previous"}
        />
        <ExploreButton
          style={{ paddingTop: "10px" }}
          onClick={this.onNextClick}
          title={"Next -->"}
        />
      </div>
    );
  }

  render() {
    return (
      <div>
        {this._renderHeader()}
        <div style={{ marginLeft: "200px", marginRight: "200px" }}>
          {this._renderTopBlurb()}
          {this._renderSelectionBox()}
          {this._renderProgressionButtons()}
        </div>
      </div>
    );
  }
}

const titleStyle = {};

const descStyle = {};

const selectionBoxStyle = {
  justifyContent: "center",
  alignItems: "center",
  background: "#F4F4F4"
};

const btnsStyle = {
  display: "flex",
  justifyContent: "space-between"
};

export default Explore;
