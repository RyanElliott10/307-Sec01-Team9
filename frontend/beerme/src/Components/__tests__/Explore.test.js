import React from "react";
import renderer from "react-test-renderer";
import { shallow, mount } from "enzyme";
import { BrowserRouter } from "react-router-dom";

import Explore from "../Explore";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Explore />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly with props", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Explore
          recommendedStyle={{ name: "Test", id: 0 }}
          currentPageIndex={2}
        />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly with state", () => {
  const ref = mount(
    <BrowserRouter>
      <Explore />
    </BrowserRouter>
  ).instance();

  ref.setState({
    recommendedStyle: true
  });

  const expectedState = {
    recommendedStyle: true
  };

  expect(ref.state).toEqual(expectedState);
});

it("renders correctly with clicks", () => {
  const ref = mount(
    <BrowserRouter>
      <Explore />
    </BrowserRouter>
  );

  ref
    .find("#submit-button")
    .at(0)
    .simulate("click");
});

it("handles checkbox selections correctly", () => {
  const ref = mount(
    <BrowserRouter>
      <Explore />
    </BrowserRouter>
  );

  const inputs = ref.find("#checkbox-option");
  inputs.forEach(r => r.simulate("click"));
});
