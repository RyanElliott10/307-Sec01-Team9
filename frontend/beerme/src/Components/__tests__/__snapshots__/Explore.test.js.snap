// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "marginLeft": "200px",
      "marginRight": "200px",
      "paddingBottom": "150px",
    }
  }
>
  <h1>
    Explore
  </h1>
  <h5>
    Use our unique beer exploration tool to find new and interesting beers for you!
  </h5>
  <div
    style={
      Object {
        "background": "#F4F4F4",
        "flexDirection": "row",
        "height": "400px",
        "justifyContent": "space-between",
        "marginTop": "15px",
        "paddingBottom": "5px",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "paddingTop": "15px",
        }
      }
    >
      <h2>
        Colors
      </h2>
      <h6
        style={
          Object {
            "color": "#696969",
          }
        }
      >
        Select your preferred color of beer.
      </h6>
    </div>
    <div
      className="justify-content-md-center row"
    >
      <div
        className="col"
      >
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={1}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={1}
                title=""
              >
                Light Yellow
              </label>
            </div>
          </div>
        </form>
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={2}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={2}
                title=""
              >
                Straw
              </label>
            </div>
          </div>
        </form>
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={3}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={3}
                title=""
              >
                Pale
              </label>
            </div>
          </div>
        </form>
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={4}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={4}
                title=""
              >
                Gold
              </label>
            </div>
          </div>
        </form>
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={5}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={5}
                title=""
              >
                Light Amber
              </label>
            </div>
          </div>
        </form>
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={6}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={6}
                title=""
              >
                Amber
              </label>
            </div>
          </div>
        </form>
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={7}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={7}
                title=""
              >
                Medium Amber
              </label>
            </div>
          </div>
        </form>
      </div>
      <div
        className="col"
      >
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={8}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={8}
                title=""
              >
                Copper
              </label>
            </div>
          </div>
        </form>
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={9}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={9}
                title=""
              >
                Light Brown
              </label>
            </div>
          </div>
        </form>
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={10}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={10}
                title=""
              >
                Saddle Brown
              </label>
            </div>
          </div>
        </form>
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={11}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={11}
                title=""
              >
                Brown
              </label>
            </div>
          </div>
        </form>
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={12}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={12}
                title=""
              >
                Dark Brown
              </label>
            </div>
          </div>
        </form>
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={13}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={13}
                title=""
              >
                Black
              </label>
            </div>
          </div>
        </form>
      </div>
    </div>
    <p
      style={
        Object {
          "alignItems": "center",
          "color": "#d1823d",
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
      Please select one option on each page before submitting.
    </p>
  </div>
  <div
    className="btn-toolbar"
    role="toolbar"
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
        "paddingTop": "10px",
      }
    }
  >
    <div />
    <div
      className="react-carousel-dots-holder "
      style={
        Object {
          "height": 10,
          "width": 60,
        }
      }
    >
      <div
        className="dot-holder"
        style={
          Object {
            "height": 10,
            "marginLeft": 1,
            "marginRight": 1,
            "width": 10,
          }
        }
      >
        <div
          className="react-carousel-dots-dot
                      
                      active"
        />
      </div>
      <div
        className="dot-holder"
        style={
          Object {
            "height": 10,
            "marginLeft": 1,
            "marginRight": 1,
            "width": 10,
          }
        }
      >
        <div
          className="react-carousel-dots-dot
                      
                      "
        />
      </div>
      <div
        className="dot-holder"
        style={
          Object {
            "height": 10,
            "marginLeft": 1,
            "marginRight": 1,
            "width": 10,
          }
        }
      >
        <div
          className="react-carousel-dots-dot
                      
                      "
        />
      </div>
    </div>
    <button
      className="submit-button btn btn-secondary"
      disabled={false}
      id="submit-button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#CC752A",
          "outline": "none",
        }
      }
      type="button"
    >
      Next
    </button>
  </div>
</div>
`;

exports[`renders correctly with props 1`] = `
<div
  style={
    Object {
      "marginLeft": "200px",
      "marginRight": "200px",
      "paddingBottom": "150px",
    }
  }
>
  <h1>
    Explore
  </h1>
  <h5>
    Use our unique beer exploration tool to find new and interesting beers for you!
  </h5>
  <div
    style={
      Object {
        "background": "#F4F4F4",
        "flexDirection": "row",
        "height": "400px",
        "justifyContent": "space-between",
        "marginTop": "15px",
        "paddingBottom": "5px",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "paddingTop": "15px",
        }
      }
    >
      <h2>
        Colors
      </h2>
      <h6
        style={
          Object {
            "color": "#696969",
          }
        }
      >
        Select your preferred color of beer.
      </h6>
    </div>
    <div
      className="justify-content-md-center row"
    >
      <div
        className="col"
      >
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={1}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={1}
                title=""
              >
                Light Yellow
              </label>
            </div>
          </div>
        </form>
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={2}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={2}
                title=""
              >
                Straw
              </label>
            </div>
          </div>
        </form>
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={3}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={3}
                title=""
              >
                Pale
              </label>
            </div>
          </div>
        </form>
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={4}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={4}
                title=""
              >
                Gold
              </label>
            </div>
          </div>
        </form>
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={5}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={5}
                title=""
              >
                Light Amber
              </label>
            </div>
          </div>
        </form>
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={6}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={6}
                title=""
              >
                Amber
              </label>
            </div>
          </div>
        </form>
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={7}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={7}
                title=""
              >
                Medium Amber
              </label>
            </div>
          </div>
        </form>
      </div>
      <div
        className="col"
      >
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={8}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={8}
                title=""
              >
                Copper
              </label>
            </div>
          </div>
        </form>
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={9}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={9}
                title=""
              >
                Light Brown
              </label>
            </div>
          </div>
        </form>
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={10}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={10}
                title=""
              >
                Saddle Brown
              </label>
            </div>
          </div>
        </form>
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={11}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={11}
                title=""
              >
                Brown
              </label>
            </div>
          </div>
        </form>
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={12}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={12}
                title=""
              >
                Dark Brown
              </label>
            </div>
          </div>
        </form>
        <form
          className=""
        >
          <div
            className="mb-3"
          >
            <div
              className="custom-control custom-checkbox"
              style={
                Object {
                  "paddingLeft": "200px",
                }
              }
            >
              <input
                checked={false}
                className="custom-control-input"
                disabled={false}
                id={13}
                md="auto"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor={13}
                title=""
              >
                Black
              </label>
            </div>
          </div>
        </form>
      </div>
    </div>
    <p
      style={
        Object {
          "alignItems": "center",
          "color": "#d1823d",
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
      Please select one option on each page before submitting.
    </p>
  </div>
  <div
    className="btn-toolbar"
    role="toolbar"
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
        "paddingTop": "10px",
      }
    }
  >
    <div />
    <div
      className="react-carousel-dots-holder "
      style={
        Object {
          "height": 10,
          "width": 60,
        }
      }
    >
      <div
        className="dot-holder"
        style={
          Object {
            "height": 10,
            "marginLeft": 1,
            "marginRight": 1,
            "width": 10,
          }
        }
      >
        <div
          className="react-carousel-dots-dot
                      
                      active"
        />
      </div>
      <div
        className="dot-holder"
        style={
          Object {
            "height": 10,
            "marginLeft": 1,
            "marginRight": 1,
            "width": 10,
          }
        }
      >
        <div
          className="react-carousel-dots-dot
                      
                      "
        />
      </div>
      <div
        className="dot-holder"
        style={
          Object {
            "height": 10,
            "marginLeft": 1,
            "marginRight": 1,
            "width": 10,
          }
        }
      >
        <div
          className="react-carousel-dots-dot
                      
                      "
        />
      </div>
    </div>
    <button
      className="submit-button btn btn-secondary"
      disabled={false}
      id="submit-button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#CC752A",
          "outline": "none",
        }
      }
      type="button"
    >
      Next
    </button>
  </div>
</div>
`;
